docs/en/sql-reference/00-sql-reference/98-sql-scripting.md

```diff
@@ -0,0 +1,108 @@
+---
+title: SQL Scripting
+---
+
+This page outlines the SQL scripting options available in Databend. You can use SQL scripting with either of the following query methods:
+
+- [Stored Procedure](/guides/query/stored-procedure)
+- [EXECUTE IMMEDIATE](/sql/sql-commands/administration-cmds/execute-immediate)
+
+### Variable Declaration
+
+Variables can be declared using the `LET` keyword, followed by the variable name, an optional type, and the initial value.
+
+```sql title='Examples:'
+LET x := 100;
+```
+
+### Query Execution
+
+SQL queries can be executed within the script, and results can be stored in variables or result sets.
+
+```sql title='Examples:'
+LET result RESULTSET := SELECT * FROM t1;
```

Comment:
All statements can include variable in scripting, using a `:`prefix to escape:

```
LET x := 1;
RETURN TABLE(SELECT :x + 1);
```

also, use `IDENTIFIER(:x)` to escape to a ident:

```
LET x := 't1';
DELETE FROM IDENTIFIER(:x);
```